src/acp/rule.ts

```diff
@@ -233,3 +247,266 @@ export function setForbiddenRuleOnPolicy(policy: Policy, rule: Rule): Policy {
   return getIriAll(policy, acp.noneOf);
 }
+
+/**
+ * ```{note} There is no Access Control Policies specification yet. As such, this
+ * function is still experimental and subject to change, even in a non-major release.
+ * ```
+ *
+ * Initialise a new empty [[SolidDataset]] to store [[Rule]]'s in.
+ */
+export const createRuleDataset = createSolidDataset;
+
+/**
+ * ```{note} There is no Access Control Policies specification yet. As such, this
+ * function is still experimental and subject to change, even in a non-major release.
+ * ```
+ *
+ * Initialise a new, empty [[Rule]].
+ *
+ * @param url URL that identifies this [[Rule]].
+ */
+export function createRule(url: Url | UrlString): Rule {
+  const stringUrl = internal_toIriString(url);
+  let ruleThing = createThing({ url: stringUrl });
+  ruleThing = setUrl(ruleThing, rdf.type, acp.Rule);
+  return ruleThing;
+}
+
+/**
+ * ```{note} There is no Access Control Policies specification yet. As such, this
+ * function is still experimental and subject to change, even in a non-major release.
+ * ```
+ *
+ * Get the [[Rule]] with the given URL from an [[RuleDataset]].
+ *
+ * @param ruleResource The Resource that contains the given [[Rule]].
+ * @param url URL that identifies this [[Rule]].
+ * @returns The requested [[Rule]], if it exists, or `null` if it does not.
+ */
+export function getRule(
+  ruleResource: RuleDataset,
+  url: Url | UrlString
+): Rule | null {
+  const foundThing = getThing(ruleResource, url);
+  if (foundThing === null || getUrl(foundThing, rdf.type) !== acp.Rule) {
+    return null;
+  }
+  return foundThing;
+}
+
+/**
+ * ```{note} There is no Access Control Policies specification yet. As such, this
+ * function is still experimental and subject to change, even in a non-major release.
+ * ```
+ *
+ * List all the agents a [[Rule]] applies **directly** to. This will not include agents
+ * that are part of a group the [[Rule]] applies to, nor will it include specific agent
+ * classes, such as authenticated or public agents.
+ *
+ * @param rule The rule from which agents are read.
+ * @returns A list of the WebIDs of agents included in the rule.
+ * @since Unreleased
+ */
+export function getAgentForRuleAll(rule: Rule): WebId[] {
+  return getIriAll(rule, acp.agent).filter(
+    (agent: WebId) =>
+      agent !== acp.PublicAgent && agent !== acp.AuthenticatedAgent
+  );
+}
+
+/**
+ * ```{note} There is no Access Control Policies specification yet. As such, this
+ * function is still experimental and subject to change, even in a non-major release.
+ * ```
+ *
+ * Overwrite the agents the [[Rule]] applies to with the provided agents.
+ *
+ * @param rule The rule for which agents are set.
+ * @param agents The list of agents the rule should apply to.
+ * @returns A copy of the input rule, applying to a different set of agents.
+ * @since Unreleased
+ */
+export function setAgentInRule(rule: Rule, agent: WebId): Rule {
```

Comment:
Yeah, this should be named `setAgentForRule()`.

Reply:
I aligned all the names to the template setXForY in 9107fe466f0f0d056bdb8146e6e14f59347f0213, and we can have a deeper naming conversation as part of the dedicted ticket this sprint.